packages/devtools_app/lib/src/memory/memory_heap_tree_view.dart

```diff
@@ -367,18 +395,50 @@ class HeapTreeViewState extends State<HeapTree>
       padding: const EdgeInsets.only(top: denseRowSpacing),
       child: Column(
         children: [
+          const SizedBox(height: defaultSpacing),
           Row(
             children: [
-              _buildSnapshotControls(textTheme),
-              const Expanded(child: SizedBox(width: defaultSpacing)),
-              // TODO(peterdjlee): Implement filter and search functionality for treemap.
+              TabBar(
+                labelColor: themeData.textTheme.bodyText1.color,
+                isScrollable: true,
+                controller: tabController,
+                tabs: HeapTreeViewState.DartHeapTabs,
+              ),
+              const Expanded(child: SizedBox()),
               _buildSearchFilterControls(),
             ],
           ),
-          const SizedBox(height: denseRowSpacing),
+          const SizedBox(height: defaultSpacing),
           Expanded(
-            child: OutlineDecoration(
-              child: buildSnapshotTables(snapshotDisplay),
+            child: TabBarView(
+              physics: defaultTabBarViewPhysics,
+              controller: tabController,
+              children: [
+                // Analysis Tab
+                Column(
+                  children: [
+                    _buildSnapshotControls(themeData.textTheme),
+                    Expanded(
+                      child: OutlineDecoration(
+                        child: buildSnapshotTables(snapshotDisplay),
+                      ),
+                    ),
+                  ],
+                ),
+
+                // Allocations Tab
+                Column(
+                  children: [
+                    _buildAllocationsControls(themeData),
+                    const SizedBox(height: denseSpacing),
```

Comment:
why is there a spacer here but not above in the analysis tab?

Reply:
No reason. Its gone.